File: src/main/java/org/polyfrost/colorsaturation/ColorSaturation.java
package org.polyfrost.colorsaturation;

import org.polyfrost.colorsaturation.command.SaturationCommand;
import org.polyfrost.colorsaturation.config.SaturationConfig;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import org.polyfrost.oneconfig.api.commands.v1.CommandManager;
import org.polyfrost.oneconfig.api.event.v1.EventManager;

/**
 * The entrypoint of the Example Mod that initializes it.
 *
 * @see Mod
 * @see org.polyfrost.oneconfig.api.event.v1.events.InitializationEvent
 */
@Mod(modid = ColorSaturation.MODID, name = ColorSaturation.NAME, version = ColorSaturation.VERSION)
public class ColorSaturation {

    // Sets the variables from `gradle.properties`. See the `blossom` config in `build.gradle.kts`.
    public static final String MODID = "@MOD_ID@";
    public static final String NAME = "@MOD_NAME@";
    public static final String VERSION = "@MOD_VERSION@";
    public static SaturationConfig config;

    // Register the config and commands.
    @Mod.EventHandler
    public void onInit(FMLInitializationEvent event) {
        config = new SaturationConfig();
        CommandManager.registerCommand(new SaturationCommand());
        EventManager.INSTANCE.register(new Saturation());
    }
}

File: src/main/kotlin/org/polyfrost/colorsaturation/ColorSaturation.kt
package org.polyfrost.colorsaturation

//#if FORGE
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
//#else
//$$ import net.fabricmc.api.ClientModInitializer
//#endif

import org.polyfrost.colorsaturation.command.SaturationCommand
import org.polyfrost.colorsaturation.config.SaturationConfig
import org.polyfrost.oneconfig.api.commands.v1.CommandManager
import org.polyfrost.oneconfig.api.event.v1.EventManager

//#if FORGE
@Mod(modid = ColorSaturation.ID, version = ColorSaturation.VERSION, name = ColorSaturation.NAME, modLanguageAdapter = "org.polyfrost.oneconfig.utils.v1.forge.KotlinLanguageAdapter")
//#endif
object ColorSaturation
    //#if FABRIC
    //$$ : ClientModInitializer
    //#endif
{

    const val ID = "@MOD_ID@"
    const val NAME = "@MOD_NAME@"
    const val VERSION = "@MOD_VERSION@"

    @JvmStatic
    lateinit var config: SaturationConfig
        private set

    fun initialize() {
        config = SaturationConfig()
        CommandManager.registerCommand(SaturationCommand())
        EventManager.INSTANCE.register(Saturation())
    }

    //#if FORGE
    @Mod.EventHandler
    fun onInit(e: FMLInitializationEvent) {
        initialize()
    }
    //#else
    //$$ override fun onInitializeClient() {
    //$$     initialize()
    //$$ }
    //#endif

}
